apps/storefront/src/store/cross-tab-sync.ts

```diff
@@ -0,0 +1,73 @@
+import type { PersistConfig } from 'redux-persist/es/types';
+import { KEY_PREFIX, REHYDRATE } from 'redux-persist/lib/constants';
+
+import type { AppStore } from '.';
+
+type CrossTabConfig<T> = Pick<PersistConfig<T>, 'blacklist' | 'whitelist'>;
+type StoreObj = ReturnType<AppStore['getState']>;
+type StoreObjKeys = keyof StoreObj;
+
+function isStringRecord(
+  objectToValidate: unknown,
+): objectToValidate is Record<keyof StoreObj[StoreObjKeys], string> {
+  if (typeof objectToValidate !== 'object' || objectToValidate === null) {
+    return false;
+  }
+
+  const values = Object.values(objectToValidate);
+
+  return values.every((value) => typeof value === 'string');
+}
+
+export default function crossTabSync(
```

Comment:
can we add a unit test for this function?

Reply:
That's an excellent idea, I'll add testing for this